### ISSUE TITLE
Short UP

### ISSUE THREAD

**Issue by comete-upn** [kanboard/kanboard#5573]
### Checklist

- [x] I verified that Kanboard is correctly installed
- [x] I verified that the problem does not come from a plugin
- [x] I verified that the problem is not already reported
- [x] I understand that Kanboard is in maintenance mode. It doesn't mean it's abandoned, but there is no significant feature development

### Actual behaviour

We updated our Kanboard to version 1.2.40 early October. We noticed that we couldn't remove attachments in tasks. After confirmation, an animation is played indefinitely and the file is not removed.

### Expected behaviour

The file should be removed.

### Steps to reproduce

**While using short urls**, add an attachment to task, then try to remove it.

### Solution

In `app/ServiceProvider/RouteProvider.php`

> ```diff
> -            $container['route']->addRoute('task/:task_id/file/:file_id/remove', 'TaskFileController', 'remove');
> +            $container['route']->addRoute('task/:task_id/file/:file_id/remove/:csrf_token', 'TaskFileController', 'remove');
> ```

**Comment by comete-upn** [comment]
Hello,

Thanks for checking. 

https://github.com/user-attachments/assets/f4a5e244-ab51-49e9-b51e-aa55d5f51034

I duplicated my kanboard and removed all plugins. Same result (see screengrab).

The method `findRoute` of the `Route` can not find the route for my URL `/kbtest/task/6237/file/8697/remove?csrf_token=xxx`.

It breaks on the `elseif ($route['items'][$i] !== $items[$i])` where:

- `$route['items'][$i]` = `remove`
- `$items[$i]` = `remove?csrf_token=xxx`

Could it be the web server? I'm using nginx.

**Comment by comete-upn** [comment]
Yes, it try another browser with a brand new vanilla profile, same result.

https://github.com/user-attachments/assets/99ff1be7-1582-45c9-a906-7dc404666688

**Comment by imfx77** [comment]
My 5 cents on the issue.
I normally install KB in the root dir of a web server and make it a subdomain. But recently I had a complaint with my plugins that there are issues when KB is installed as a subdir (which seems to be the recommended usage). And while fixing my plugins I hit some issues with KB itself.
```
e.g. 
https://kbtest.mysite.com/
VS
https://mysite.com/kbtest/
```

I am using nginx and tested with KB v1.2.35 and v1.2.44.
* when installed in root dir and used as subdomain both versions work as expected
* when installed in a subdir:
* * if link rewrite is not enabled (i.e. using full blown URLs) then links work OK
* * if using short links through routes then clicking a link gives 404 on both version - I was only able to fix this by explicitly configuring the web server to redirect 404 error to ```https://mysite.com/kbtest/index.php```. However, this means that if I have several instances of KB in different subfolders kbtest1/2/3, only one of them will work OK at a time.

Another, not route related issue that I hit for subfolder installs is that log in and out don't work or work quite weird:
* for v1.2.35
* * after login some random/weird redirection happens (in my case loading a JS file of some plugin) and it is necessary to manually re-enter the URL in the browsers address bar to actually see the user's dashboard
* * log out just gives a blank white page, then again manually reloading brings up the login screen
* for v.1.2.44
* * log out gives 403, so once logged into an account you can never log out and log in as a different user

Hope this info helps to narrow down the issues 🤞
